sirepo/package_data/static/js/raydata.js

```diff
@@ -1147,6 +1165,91 @@ SIREPO.app.directive('columnList', function() {
 });
 
+SIREPO.app.directive('scanDetail', function() {
+    return {
+        restrict: 'A',
+        scope: {
+            scan: '<',
+        },
+        template: `
+            <div><strong>Scan Detail</strong></div>
+            <div class="well" style="height: 250px; overflow: auto;">
+            <div data-ng-if="scan">
+              <div><strong>Scan Id:</strong> {{ scan.rduid }}</div>
+              <div data-ng-if="detailedStatusFile()">
+                <div><strong>Most Recent Status</strong></div>
+                <pre>{{ currentStatus() }}</pre>
+              </div>
+              <div data-ng-if="detailedStatusFile()">
+                <div><strong>Detailed Status File</strong></div>
+                <pre>{{ detailedStatus() }}</pre>
+              </div>
+              <div data-ng-if="detailedStatusFile()">
+                <div><strong>Current Consecutive Failures:</strong> {{ consecutiveFailures() }}</div>
+              </div>
+              <div data-ng-if="analysisElapsedTime()"><strong>Analysis Elapsed Time:</strong> {{ analysisElapsedTime() }} seconds</div>
+            </div>
+            </div>
+`,
+        controller: function($scope, columnsService, utilities) {
+            function failureInRun(run) {
+                let r = false;
+                for (const f of Object.values($scope.detailedStatusFile()[run])) {
+                    if (f.status === 'failed') {
+                        r = true;
+                    }
+                }
+                return r;
+            }
+
+            function getSortedRunIndexes() {
+                return Object.keys($scope.detailedStatusFile()).map((x) => parseInt(x)).sort();
+            }
+
+            function mostRecentAnalysisDetails() {
+                return $scope.detailedStatusFile()? $scope.detailedStatusFile()[Math.max(...getSortedRunIndexes())] : '';
+             }
+
+            $scope.analysisElapsedTime = () => {
+                return $scope.scan && $scope.scan.analysis_elapsed_time ? $scope.scan.analysis_elapsed_time : null;
+            };
+
+            $scope.consecutiveFailures = () => {
+                if (! $scope.detailedStatusFile()) {
+                    return '';
+                }
+                let r = 0;
+                for (const k of getSortedRunIndexes().reverse()) {
+                    if (failureInRun(k)) {
+                        r += 1;
+                    } else {
+                        return r;
+                    }
+                }
+
+                return r;
+            };
+
+            $scope.currentStatus = () => {
+                let r = '';
+                for (const k of Object.keys(mostRecentAnalysisDetails())) {
+                    r += k + ': ' + mostRecentAnalysisDetails()[k].status + '\n';
+                }
+                return r;
+            };
+
+            $scope.detailedStatus = () => {
+                return utilities.objectToText($scope.detailedStatusFile());
+            };
+
```

Comment:
I made one more change - the status panel was overloaded with scrollbars, including a horizontal scrollbar which was shown but not used:
![x](https://github.com/radiasoft/sirepo/assets/6516910/a5adff01-f9f0-457e-ab3e-ae9379bd8262)

So I removed the inner scrollbars and set the outer scrollbar to "auto":
![x2](https://github.com/radiasoft/sirepo/assets/6516910/baad3846-a979-43b5-9b24-c84862640d99)

I also formatted the json in the detail status panel to remove the unneeded syntax.

Reply:
Thanks, looks good. Interestingly the horizontal scrollbars weren't showing for me on any browser